che.sh

```diff
@@ -740,38 +740,66 @@ update_che_image() {
+    WS_PORT=${1}
 
-  if [ ! -d "${MOUNT_PATH}" ]; then
-    error "${CHE_MINI_PRODUCT_NAME} mount: Path provided is not a valid directory."
-    return
+  # Port not provided, let's do a simple discovery of running workspaces
+  else 
+    info "${CHE_MINI_PRODUCT_NAME} mount: Searching for running workspaces with open SSH port..."
+
+    CURRENT_WS_INSTANCES=$(docker ps -aq --filter "name=_che_dev-machine")
+    CURRENT_WS_COUNT=$(echo $CURRENT_WS_INSTANCES | wc -w)
+    
+    if [ $CURRENT_WS_COUNT -eq 0 ]; then
+      error "${CHE_MINI_PRODUCT_NAME} mount: We could not find any running workspaces"
+
+    elif [ $CURRENT_WS_COUNT -eq 1 ]; then
+      RUNNING_WS_PORT=$(docker inspect --format='{{ (index (index .NetworkSettings.Ports "22/tcp") 0).HostPort }}' ${CURRENT_WS_INSTANCES})
+      info "${CHE_MINI_PRODUCT_NAME} mount: Connecting to remote workspace on port $RUNNING_WS_PORT"
+      WS_PORT=$RUNNING_WS_PORT
+   
+    else 
+      info "${CHE_MINI_PRODUCT_NAME} mount: Discovered $CURRENT_WS_COUNT running workspaces with SSH ports"
+      info "${CHE_MINI_PRODUCT_NAME} mount: Choose one and re-run with 'che mount <ws-port>'"
+      info "${CHE_MINI_PRODUCT_NAME} mount: Available workspace ports:"
+      IFS=$'\n'
+      for CHE_WS_CONTAINER_ID in $CURRENT_WS_INSTANCES; do
+        CURRENT_WS_PORT=$(docker inspect --format='{{ (index (index .NetworkSettings.Ports "22/tcp") 0).HostPort }}' ${CHE_WS_CONTAINER_ID})
+        echo $CURRENT_WS_PORT
+      done
+      return
+    fi
   fi
   
   if is_native; then
-    PLATFORM_SPECIFIC_OPTIONS="-v ${HOME}/.ssh:${HOME}/.ssh \
-            -v ${HOME}/.unison:${HOME}/.unison \
-            -v /etc/group:/etc/group:ro \
-            -v /etc/passwd:/etc/passwd:ro \
-            -u $(id -u ${USER})"
+    docker_run_with_che_properties --cap-add SYS_ADMIN \
+                                   --device /dev/fuse \
+                                   -v ${HOME}/.ssh:${HOME}/.ssh \
+                                   -v ${HOME}/.unison:${HOME}/.unison \
+                                   -v /etc/group:/etc/group:ro \
+                                   -v /etc/passwd:/etc/passwd:ro \
+                                   -u $(id -u ${USER})
+                                   -v "${MOUNT_PATH}":/mnthost \
+                                   "${CHE_MOUNT_IMAGE_NAME}":"${CHE_UTILITY_VERSION}" \
+                                        "${GLOBAL_GET_DOCKER_HOST_IP}" $WS_PORT
+    
   else
-    PLATFORM_SPECIFIC_OPTIONS="-v ${HOME}/.ssh:/root/.ssh"
+    docker_run_with_che_properties --cap-add SYS_ADMIN \
+                                   --device /dev/fuse \
+                                   -v "${HOME_PATH}"/.ssh:/root/.ssh \
+                                   -v "${MOUNT_PATH}":/mnthost \
+                                   "${CHE_MOUNT_IMAGE_NAME}":"${CHE_UTILITY_VERSION}" \
+                                        "${GLOBAL_GET_DOCKER_HOST_IP}" $WS_PORT
```

Comment:
The indent was intended that way for readability - because it's the post container params.